pywbemtools/pywbemcli/_cmd_statistics.py

```diff
@@ -0,0 +1,436 @@
+
+    click.echo("Server GatherStatisticalData set {}".
+               format(to_on_off(desired_state)))
+
+
+###############################################################
+#         Statistics cmds action functions
+###############################################################
+
+
+def cmd_statistics_server_on(context):
+    """
+    Attempt to enable the statistics gathering on the current server if there
+    is a current server defined.
+    """
+    test_conn_exists(context)
+    set_server_statistics(context, True)
+
+
+def cmd_statistics_server_off(context):
+    """
+    Attempt to disable the statistics gathering on the current server if there
+    is a current server defined.
+    """
+    test_conn_exists(context)
+    set_server_statistics(context, False)
+
+
+def cmd_statistics_status(context):
+    """
+    Show current status of both client and WBEM server statistics if there
+    is a connection.
+
+    It gets the status of the WBEM server statistics from the WBEM server
+    and the client statistics from the context and displays either a
+    table or text representation of the status.
+
+    If there is an issue with getting statistics information from the server,
+    it displays detailed information about the server response.
+
+    """
+
+    # validate output format with default of table
+    output_format = validate_output_format(context.output_format, ['TABLE',
+                                                                   'TEXT'])
+
+    test_conn_exists(context)
+
+    # Display server status as on/off
+    try:
+        objmgr = get_objmgr_inst(context)
+        try:
+            svr_status = objmgr[OBJMGR_STAT_PROPERTY_NAME]
+            svr_status = to_on_off(svr_status)
+        except KeyError:
+            svr_status = "No property {}".format(OBJMGR_STAT_PROPERTY_NAME)
+    except Error as er:
+        svr_status = "Not settable {}".format(er)
+
+    context.spinner_stop()
+
+    pywbemcli_stats_status = to_on_off(context.timestats)
+    if output_format_is_table(output_format):
+        headers = ['statistics source', 'status']
+
+        rows = [["client statistics", pywbemcli_stats_status],
+                ['server statistics', svr_status]]
+        click.echo(format_table(rows, headers, "Statistics status"))
+
+    else:
+        click.echo("Statistics status: client statistics={}; Server "
+                   "statistics={}".format(pywbemcli_stats_status,
+                                          svr_status))
+
+
+def cmd_statistics_reset(context):
+    """
+    Reset the client statistics. This command does not impact the
+
+    statistics retrieved from the server with ``statistics server-show``.
+    It does reset the statistics kept in the pywbem client for server
+    response time.
+    """
+    test_conn_exists(context)
+
+    context.conn.statistics.reset()
+
+    context.spinner_stop()
+    click.echo("Statistics reset")
```

Comment:
In the "format_statistics()" method in "_context_obj.py", the title could also be made explicit in that it displays the client statistics. Also, the explanations in the title line seem unnecessary to me:

Current code:
```
        title = 'Statistics: Time(Seconds); Times/lengths: (Avg/Min/Max), ' \
                'single value if all same.'
```
Proposed code:
```
        title = 'Client statistics:'
```

Reply:
DONE